src/main/java/ui/components/pickers/MilestonePickerDialog.java

```diff
@@ -43,6 +43,20 @@ public MilestonePickerDialog(Stage stage, TurboIssue issue, List<TurboMilestone>
         state = new MilestonePickerState(originalMilestones);
         initUI();
         setupKeyEvents();
+        fillInputField();
+    }
+
+    /**
+     * Fills the input field of milestone picker dialog
+     * If there is an existing milestone, input field is filled with existing milestone's title
+     * Else input field is filled with the next open milestone that is not overdue
+     */
+    private void fillInputField() {
+        if (!getExistingMilestone(originalMilestones).isPresent()) {
+            getNextOpenMilestone(originalMilestones)
+                    .map(PickerMilestone::getTitle)
+                    .ifPresent(this::fillInputFieldWithMilestoneName);
+        }
```

Comment:
To have better SLAP, maybe abstract the first line to another method `getDefaultMilestone`. Again, a `static` method on `PickerMilestone`. If you can, write tests for the methods, at the very least for `getNextOpenMilestone` since it is introduced in this PR.